STRINGIFY does not work with mcumgr

**Describe the bug**
A clear and concise description of what the bug is.

I've recently done a git pull and west update to the latest.

I have a compilation error:

```
In file included from /home/vince/zephyrproject/zephyr/subsys/mgmt/mcumgr/lib/cmd/img_mgmt/src/img_mgmt_state.c:16:
/home/vince/zephyrproject/zephyr/subsys/mgmt/mcumgr/lib/cmd/img_mgmt/src/img_mgmt_state.c: In function 'img_mgmt_state_write':
/home/vince/zephyrproject/zephyr/subsys/mgmt/mcumgr/lib/util/include/zcbor_bulk/zcbor_bulk_priv.h:37:13: warning: implicit declaration of function 'STRINGIFY' [-Wimplicit-function-declaration]
   37 |    .value = STRINGIFY(k),   \
      |             ^~~~~~~~~
/home/vince/zephyrproject/zephyr/subsys/mgmt/mcumgr/lib/cmd/img_mgmt/src/img_mgmt_state.c:276:3: note: in expansion of macro 'ZCBOR_MAP_DECODE_KEY_VAL'
  276 |   ZCBOR_MAP_DECODE_KEY_VAL(hash, zcbor_bstr_decode, &zhash),
      |   ^~~~~~~~~~~~~~~~~~~~~~~~
/home/vince/zephyrproject/zephyr/subsys/mgmt/mcumgr/lib/cmd/img_mgmt/src/img_mgmt_state.c:276:28: error: 'hash' undeclared (first use in this function); did you mean 'zhash'?
  276 |   ZCBOR_MAP_DECODE_KEY_VAL(hash, zcbor_bstr_decode, &zhash),
      |                            ^~~~
```

Well, OK, fine, we have no STRINGIFY() so the macro expansion does not work because the macro is trying to turn those bare words like "hash" into a string "hash" and it can't because it has no STRINGIFY.

`subsys/mgmt/mcumgr/lib/cmd/img_mgmt/src/img_mgmt.c`

calls that ZCBOR_MAP_DECODE_KEY macro which is defined in:

`#include "zcbor_bulk/zcbor_bulk_priv.h"`

and that apparently does not know how to STRINGIFY so around line 13 of that file, I do know where I can find a STRINGIFY and I add:

`#include <zephyr/toolchain/common.h>`

and now compilation succeeds with no errors.

So my git diff looks like this:

```
vince@VINCE:~/zephyrproject/zephyr$ git diff
diff --git a/subsys/mgmt/mcumgr/lib/util/include/zcbor_bulk/zcbor_bulk_priv.h b/subsys/mgmt/mcumgr/lib/util/include/zcbor_bulk/zcbor_bulk_priv.h
index a15511817c..cf3647a709 100644
--- a/subsys/mgmt/mcumgr/lib/util/include/zcbor_bulk/zcbor_bulk_priv.h
+++ b/subsys/mgmt/mcumgr/lib/util/include/zcbor_bulk/zcbor_bulk_priv.h
@@ -10,6 +10,8 @@
 extern "C" {
 #endif
 
+#include <zephyr/toolchain/common.h>
+
 /** @cond INTERNAL_HIDDEN */
 
 struct zcbor_map_decode_key_val {
vince@VINCE:~/zephyrproject/zephyr$
``` 

I don't understand enough of ZCBOR_MAP_DECODE_KEY_VAL to know what's going on, but it clearly wants to STRINGIFY something so I gave it a STRINGIFY and my firmware seems to work.  I don't know if this is the "right" way to fix this; but it does compile...

Please also mention any information which could help others to understand
the problem you're facing:
 - What target platform are you using?
 - What have you tried to diagnose or workaround this issue?
 - ...

**To Reproduce**
Steps to reproduce the behavior:
1. mkdir build; cd build
2. cmake -DBOARD=board\_xyz
3. make
4. See error

**Expected behavior**
A clear and concise description of what you expected to happen.

No errors upon compilation.

**Impact**
What impact does this issue have on your progress (e.g., annoyance, showstopper)

Could not compile Zephyr until I added that include line to subsys/mgmt/mcumgr/lib/util/include/zcbor_bulk/zcbor_bulk_priv.h which may or may not be the "right" way to fix an error like that.  But it did work.

**Logs and console output**

**Environment (please complete the following information):**
Running on Win10 in a WSL running Ubuntu (long story) but I don't think STRINGIFY() not being defined would be platform dependent.

**Additional context**

Anyway best of luck and have a great day.

--- a/subsys/mgmt/mcumgr/lib/util/include/zcbor_bulk/zcbor_bulk_priv.h
+++ b/subsys/mgmt/mcumgr/lib/util/include/zcbor_bulk/zcbor_bulk_priv.h
@@ -10,6 +10,8 @@
 extern "C" {
 #endif
 
+#include <toolchain.h>
+
 /** @cond INTERNAL_HIDDEN */
 
 struct zcbor_map_decode_key_val {

